fmriprep/workflows/anatomical.py

```diff
@@ -74,10 +76,130 @@ def t1w_preprocessing(name='t1w_preprocessing', settings=None):
+                                          'acq_id', 'rec_id', 'run_id']))
+
+        bids_info = pe.Node(
+            niu.Function(function=bidsinfo, input_names=['in_file'],
+                         output_names=['subject_id', 'ses_id', 'task_id',
+                                       'acq_id', 'rec_id', 'run_id']),
+            name='BIDSInfo',
+            run_without_submitting=True)
+
+        autorecon1 = pe.Node(
+            freesurfer.ReconAll(
+                directive='autorecon1',
+                openmp=nthreads,
+                parallel=True),
+            name='Reconstruction')
+        autorecon1.interface._can_resume = False
+        autorecon1.interface.num_threads = nthreads
+        if 'FREESURFER_SUBJECTS' in os.environ:
+            autorecon1.inputs.subjects_dir = os.getenv('FREESURFER_SUBJECTS')
+
+        def inject_skullstripped(subjects_dir, subject_id, skullstripped):
+            import os
+            import nibabel as nib
+            from nilearn.image import resample_to_img, new_img_like
+            from nipype.utils.filemanip import copyfile
+            mridir = os.path.join(subjects_dir, subject_id, 'mri')
+            t1 = os.path.join(mridir, 'T1.mgz')
+            bm_auto = os.path.join(mridir, 'brainmask.auto.mgz')
+            bm = os.path.join(mridir, 'brainmask.mgz')
+
+            img = nib.load(t1)
+            mask = nib.load(skullstripped)
+            bmask = new_img_like(mask, mask.get_data() > 0)
+            resampled_mask = resample_to_img(bmask, img, 'nearest')
+            masked_image = new_img_like(img, img.get_data() * resampled_mask.get_data())
+            masked_image.to_filename(bm_auto)
+
+            copyfile(bm_auto, bm, copy=True, use_hardlink=True)
+            return subjects_dir, subject_id
+
+        injector = pe.Node(
+            niu.Function(
+                function=inject_skullstripped,
+                input_names=['subjects_dir', 'subject_id', 'skullstripped'],
+                output_names=['subjects_dir', 'subject_id']),
+            name='InjectSkullstrip')
+
+        reconall = pe.Node(
+            ReconAllRPT(
+                openmp=nthreads,
+                parallel=True,
+                out_report='reconall.svg',
+                generate_report=True),
+            name='Reconstruction2')
+        reconall.interface.num_threads = nthreads
+
+        recon_report = pe.Node(
+            DerivativesDataSink(base_directory=settings['output_dir'],
+                                suffix='reconall', out_path_base='reports'),
+            name='ReconAll_Report'
+        )
+
+        def joiner(dirname, basename):
+            import os
+            return os.path.join(dirname, basename)
+
+        subject_dir = pe.Node(
+            niu.Function(
+                function=joiner,
+                input_names=['dirname', 'basename'],
+                output_names=['path']),
+            name='SubjectDir',
+            run_without_submitting=True)
```

Comment:
Is there a better way to get the full subject directory? It's 11 lines for `os.path.join`.